pkg/client/client.go

```diff
@@ -66,28 +73,29 @@ func (c *tlsBootstrapClientImpl) setupClientConnection(ctx context.Context) (*gr
 	c.logger.Info("decoded cluster CA data")
 
 	c.logger.Debug("generating TLS config for GRPC client connection...")
-	tlsConfig, err := getTLSConfig(pemCAs, c.nextProto, execCredential.Spec.Cluster.InsecureSkipTLSVerify)
+	tlsConfig, err := GetTLSConfig(pemCAs, c.nextProto, execCredential.Spec.Cluster.InsecureSkipTLSVerify)
```

Comment:
we could theoretically mock this similarly to how we're now doing it for fileReader (e.g. create a common func type and have a field on the client of this type which we set to this func outside of tests, while within tests we use a mock function which satisfies the type